apis/datadoghq/v1alpha1/datadogagent_types.go

```diff
@@ -779,12 +779,22 @@ type NodeAgentConfig struct {
 	// +optional
 	Checksd *ConfigDirSpec `json:"checksd,omitempty"`
 
-	// Provide a mapping of Kubernetes Labels to Datadog Tags.
+	// Provide a mapping of Kubernetes Namespace Labels to Datadog Tags.
+	// <KUBERNETES_LABEL>: <DATADOG_TAG_KEY>
+	// +optional
+	NamespaceLabelsAsTags map[string]string `json:"namespaceLabelsAsTags,omitempty"`
```

Comment:
do we want this env var set in containers besides the `agent` container?

Reply:
Good catch, looks like in the Helm chart these 4 are a part of the common environment variables distributed to each container. Now _technically_ `NodeLabelsAsTags` would probably be fine to just live in the `agent` container since that is setting the host-tag. But will look at looping all of those into the other containers through that same common env setup in the operator.

Reply:
thanks for the update. i'm a bit torn because it becomes less intuitive when its config is under `NodeAgentConfig` (`agent` container) but the envvar gets applied to all containers. the existing pattern with `PodLabelsAsTags` was already "broken" in that it only applied to the `agent` container, right?

maybe for v1 we keep it as it originally was (reverting your latest commit) and for v2 we can implement this properly. what do you think?

Reply:
I was a torn about that too. You're right the current pattern for the `PodLabelsAsTags` was only applying it for the `agent` container so that was "broken" before.  

This could get moved into the parent as {`agent.podLabelsAsTags`, `agent.nodeLabelsAsTags`, ...}, or if it would be cleaner {`agent.tagExtraction.podLabelsAsTags`, `agent.tagExtraction.nodeLabelsAsTags`, ...}. Currently in the Helm Chart these are just under the `datadog` parent, so I guess the closest equivalent would be to load them under {`features.podLabelsAsTags`, `features.nodeLabelsAsTags`, ...}.

Reply:
yea, i think ideally they would be moved to the parent, but then it would introduce a breaking change. and just putting `nodeLabelsAsTags` in the parent would be confusing. (`features` is intended to be for "products", more or less.)

Reply:
For now will remain as {`agent.config.podLabelsAsTags`, `agent.config.nodeLabelsAsTags`, ...} to remain as non-breaking change. With the caveat this will _not_ add those env vars to the other containers. Will adjust in v2, as the structure is going to change anyway.